docs/RevisionHowto.md

```diff
@@ -0,0 +1,161 @@
+# Step-by-step example
```

Comment:
I think it would be useful to add a few details here: 
* what exactly is  the example demonstrating ?    (is it adding revisions for those libraries from cardano-base? 
* under what circumstanced would we need to do this? 
* what is the meaning of the commits mentioned in the code examples, eg: why  checkout  commit `4fab1fe2`? 

Also I don't understand exactly what the relationship of this paragraph is with the one at the end, `# Summary of steps to take when adding and testing revisions`  , is the latter a more abstract way of describing it , whereas the former is a concrete example? If so, maybe we can change the order, so the example would be maybe easier to follow if we knew what we are trying to achieve?

Reply:
Thanks for the detailed review! I addressed your suggestions, hopefully it is now a bit easier to read and more clear. I will add a few more things to the `Building with Nix` instructions (more info on `.#allSmokeTestPackages` and will probably replace that with the exact packages as that obviously builds much faster).